src/address/mod.rs

```diff
@@ -329,6 +339,58 @@ where
     }
 }
 
+/// A weakly referenced counterpart to `Recipient<M>`
+pub struct WeakRecipient<M: Message>
+where
+    M: Message + Send,
+    M::Result: Send,
+{
+    wtx: Box<dyn WeakSender<M> + Sync>,
+}
+
+impl<M> WeakRecipient<M>
+where
+    M: Message + Send,
+    M::Result: Send,
+{
+    pub(crate) fn new(wtx: Box<dyn WeakSender<M> + Sync>) -> WeakRecipient<M> {
+        WeakRecipient { wtx }
+    }
+
+    /// Attempts to upgrade the `WeakRecipient<M>` pointer to an `Recipient<M>`, similar to `WeakAddr<A>`
+    pub fn upgrade(self) -> Option<Recipient<M>> {
+        self.wtx.upgrade().map(Recipient::new)
+    }
+}
+
+impl<A: Actor, M: Message + Send + 'static> Into<WeakRecipient<M>> for Addr<A>
+where
+    A: Handler<M>,
+    M::Result: Send,
+    A::Context: ToEnvelope<A, M>,
+{
+    fn into(self) -> WeakRecipient<M> {
+        self.downgrade().recipient()
+    }
+}
+
+impl<A: Actor, M: Message + Send + 'static> Into<WeakRecipient<M>> for WeakAddr<A>
+where
+    A: Handler<M>,
+    M::Result: Send,
+    A::Context: ToEnvelope<A, M>,
+{
+    fn into(self) -> WeakRecipient<M> {
+        WeakRecipient::new(Box::new(self.wtx))
+    }
+}
+
+/// The `Recipient` type allows to send one specific message to an
```

Comment:
Spurious docstring (possibly a copy-paste leftover)?

Reply:
whoops, yes indeed, I was shuffling files around a bit there, after I had moved recipient and weak_recipient into their own submodules and then reintegrated them. these files contain a bunch of structs at once and tons of trait implementations, I would at some point ask if we perhaps want to split them up a little.

tanks for the find!